ksql-engine/src/main/java/io/confluent/ksql/util/LagInfoKey.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import com.google.common.base.Preconditions;
+import java.util.Objects;
+
+/**
+ * This represents a unique store in the system and the basis for getting lag information from
+ * KafkaStreams, exposed by the QueryMetadata.
+ */
+public final class LagInfoKey {
```

Comment:
That makes sense, though we're in need of some way to refer to a state store more generally.  At request time, we'll need to pick through persistent query state in order to create this, which isn't easy since it's kind of hidden.  It might make sense to add some method `getStateStoreId` which returns this kind of object from one of the persistent query data structures (and has access to internal state like state store name), and generalizes it a bit instead of making this key lag related.  What do you think?  Also, which persistent query data structure would own that `StateStoreId`?

Reply:
I made a class QueryStateStoreId as a rest entity since it's used as part of the API and has jackston annotations.  Tell me if you have another recommendation here.